uitools/cpp/Esri/ArcGISRuntime/Toolkit/OverviewMapController.cpp

```diff
@@ -0,0 +1,395 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "OverviewMapController.h"
+
+// Toolkit headers
+#include "Internal/GeoViews.h"
+
+// ArcGISRuntime headers
+#include <Graphic.h>
+#include <GraphicsOverlay.h>
+#include <Map.h>
+#include <MapTypes.h>
+#include <SimpleFillSymbol.h>
+#include <SimpleLineSymbol.h>
+#include <SimpleMarkerSymbol.h>
+#include <Viewpoint.h>
+
+namespace Esri {
+namespace ArcGISRuntime {
+namespace Toolkit {
+
+  namespace {
+
+    /*!
+      \internal
+      Connect to MapView's viewpointChanged. Note, we display the visible area as the
+      graphic, and we apply viewpoint rotations.
+     */
+    void connectToMapView(OverviewMapController* self, MapViewToolkit* view, Graphic* reticle)
+    {
+      QObject::connect(view, &MapViewToolkit::viewpointChanged, self,
+                       [self, view, reticle]
+                       {
+                         reticle->setGeometry(view->visibleArea());
+                         if (view->isNavigating())
+                         {
+                           const Viewpoint v = view->currentViewpoint(ViewpointType::CenterAndScale);
+                           Viewpoint newView{
+                               geometry_cast<Point>(v.targetGeometry()),
+                               v.targetScale() * self->scaleFactor(),
+                               v.rotation()};
+                           self->insetView()->setViewpoint(newView, 0);
+                         }
+                       });
+    }
+
+    /*!
+      \internal
+      Connect to SceneView's viewpointChanged. Note, we display centre and scale as the
+      graphic, and we ignore viewpoint rotations.
+     */
+    void connectToSceneView(OverviewMapController* self, SceneViewToolkit* view, Graphic* reticle)
+    {
+      QObject::connect(view, &SceneViewToolkit::viewpointChanged, self,
+                       [self, view, reticle]
+                       {
+                         const Viewpoint v = view->currentViewpoint(ViewpointType::CenterAndScale);
+                         reticle->setGeometry(v.targetGeometry());
+                         if (view->isNavigating())
+                         {
+                           Viewpoint newView{
+                               geometry_cast<Point>(v.targetGeometry()),
+                               v.targetScale() * self->scaleFactor()};
+                           self->insetView()->setViewpoint(newView, 0);
```

Comment:
```suggestion
                               v.targetScale() * self->scaleFactor()};
                           constexpr float animationDuration {0};
                           self->insetView()->setViewpoint(newView, animationDuration);
```

Optional. Since this is internal you don't need to do this, but it helps the reader a bit.

There are a few of these, so optional throughout. I will defer to your judgement.